core/views.py

```diff
@@ -147,4 +148,69 @@ class CommunityDetailView(DetailView):
     context_object_name = "community"
 
     def get_object(self: "CommunityDetailView") -> Community:
-        return Community.objects.get(slug=self.kwargs["slug"])
+        community = Community.objects.get(slug=self.kwargs["slug"])
+        if community.privacy == "PRIVATE" and not community.members.filter(id=self.request.user.id).exists():
+            raise PermissionDenied
+        return community
+
+    def get_context_data(self: "CommunityDetailView", **kwargs: any) -> dict[str, any]:
+        context = super().get_context_data(**kwargs)
+        community = self.get_object()
+        user = self.request.user
+
+        if user.is_authenticated:
+            context["is_admin_or_moderator"] = community.is_admin_or_moderator(user)
+        else:
+            context["is_admin_or_moderator"] = False
+
+        context["add_moderator_form"] = AddModeratorForm()
```

Comment:
I am thinking this forms should be only available for users that are
- authenticated
- and creators or admins
Otherwise it is unnecessary form on page even if not working.